haskell/private/path_utils.bzl

```diff
@@ -79,13 +79,13 @@ def declare_compiled(hs, src, ext, directory=None):
     hs: Haskell context.
     src: Haskell source file.
     ext: New extension.
-    directory: Directory the new file should live in.
+    directory: String, directory prefix the new file should live in.
 
   Returns:
     File: Declared output file living in `directory` with given `ext`.
   """
   fp = paths.replace_extension(_rel_path_to_module(hs, src), ext)
-  fp_with_dir = fp if directory == None else paths.join(directory.basename, fp)
+  fp_with_dir = fp if directory == None else paths.join(directory, fp)
```

Comment:
I didn't see anywhere we're calling this function with `directory=None`.  Can we remove that logic?

Reply:
This allows to omit the `directory` argument, may be useful in future even though we don't use it right now (used in the past though).